drivers/gpu/drm/drm_gem_ttm_helper.c

```diff
@@ -4,6 +4,11 @@
 
 #include <drm/drm_gem_ttm_helper.h>
 
+#ifdef __FreeBSD__
+/* Workaround for circular drm<->ttm dependency */
+extern __weak int ttm_bo_mmap_obj(struct vm_area_struct *vma, struct ttm_buffer_object *bo);
```

Comment:
I wonder how the circular dependency just works on Linux

Reply:
I have no idea, same problem exists with vram_helpers

Reply:
Oh! it's probably a separate module, this file has `MODULE_DESCRIPTION("DRM gem ttm helpers");` at the bottom!

Well, a weak symbol seems like a good solution for avoiding extra modules...??

Reply:
Yeah, I'm pretty sure they don't work across modules, but this shouldn't cause any issues with amdgpu (unless just having a weak symbol in the module screws something up). `drm_gem_vram_helper.c` (which is the only consumer of `drm_gem_ttm_mmap` which is what has the dependency) is currently included in `vboxvideo/Makefile` only.

Huh, so these helpers have been included in our makefiles for the drivers themselves? Okay I guess we can just do the same for `drm_gem_ttm_helper.c`